.github/workflows/generate-preview-link.yml

```
@@ -0,0 +1,163 @@
+name: Generate preview link
+
+permissions:
+    actions: write
+    checks: write
+    contents: write
+    deployments: write
+    pull-requests: write
+    statuses: write
+
+on:
+    workflow_run:
+        workflows: ['Pre-generate preview link']
+        types:
+            - completed
+
+env:
+    NODE_OPTIONS: '--max-old-space-size=8192'
+
+concurrency:
+    group: cloudflare-pages-build-${{ github.event.workflow_run.head_branch }}
+    cancel-in-progress: true
```

Seems like this will cancel concurrent build, can we put through some kind of queue?
```suggestion
    cancel-in-progress: true
```

---

Yeap that's the plan, when the previous build is running and a new build is re-triggered (either due to a new commit to the PR), we cancel the previous build and re-run the build since the previous workflow will contain the build without the new commit changes. Additionally this also avoids waiting for the previous workflow to finish since its redundant anyway.

---

And this will not cancel other workflows for other pull requests, because its concurrent group is uniquely identified by the pull request head branch:
```
cloudflare-pages-build-${{ github.event.workflow_run.head_branch }}
```